library/include/rocblas-auxiliary.h

```diff
@@ -19,7 +19,7 @@
 /* ============================================================================================ */
 /*! \brief  indicates whether the pointer is on the host or device. currently HIP API can only recoginize the input ptr on deive or not
 can not recoginize it is on host or not */
-ROCBLAS_EXPORT rocblas_mem_location
+ROCBLAS_EXPORT rocblas_pointer_mode
 rocblas_get_pointer_location(void *ptr);
```

Comment:
An alternative would be to have two enums 1) rocblas_pointer_mode   2) rocblas_mem_location  . rocblas_pointer_mode would be set by the user to indicate if their pointers are on host or device. This is the same as the use in cuBLAS.  rocblas_mem_location would have its current usage, that is, the function would check where the pointer is, and use this to call the appropriate device kernel.